### PR TITLE
WIP: Implement assert a file matches a fixture on disk

### PR DESCRIPTION
**Pull request by farodin91**

### CODE REVIEW

File: src/path/ft.rs [hunk 1]
```diff
@@ -59,6 +72,17 @@ impl FileTypePredicate {
         self.follow = yes;
         self
     }
+
+    pub(crate) fn new(path: &path::Path) -> io::Result<FileTypePredicate> {
```
Comment (epage) [inline src/path/ft.rs]: Would it be useful to make this `pub`?

What about naming it `from_path` or `with_path`?

File: src/path/ft.rs [hunk 2]
```diff
@@ -59,6 +72,17 @@ impl FileTypePredicate {
...
+        Ok(FileTypePredicate {
+            ft: FileType::new(path)?,
+            follow: true,
+        })
+    }
+
+    pub(crate) fn is_file(&self) -> bool {
```
Comment (epage) [inline src/path/ft.rs]: I think I'd rather we just return the file type.

File: src/path/fs.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+impl fmt::Display for StrFilePredicate {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "content is {}", self.content)
+    }
+}
+
+impl Predicate<path::Path> for StrFilePredicate {
+    fn eval(&self, path: &path::Path) -> bool {
+        let content = FileContent::new(path).unwrap().utf8();
```
Comment (epage) [inline src/path/fs.rs]: Please address `unwrap`

File: src/path/fs.rs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+/// Predicate that compares string content of files
+#[derive(Debug)]
+pub struct StrFilePredicate {
+    content: String,
+}
+
+impl fmt::Display for StrFilePredicate {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "content is {}", self.content)
```
Comment (epage) [inline src/path/fs.rs]: 1. this should be `var is {}`
2. Should we print `content` or path?

File: src/path/fs.rs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+use std::fmt;
+use std::path;
+
+use path::{FileContent, FileTypePredicate};
+use Predicate;
+
+/// Predicate that compares file matches
+#[derive(Clone, Debug)]
+pub struct FileSystemEntryPredicate {
```
Comment (epage) [inline src/path/fs.rs]: Should we make this only work with files?  We can handle dirs like we handle symlinks or non-existent files

If we keep it handling dirs, maybe shorten the name to `EntryPredicate`?

File: src/path/fs.rs [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    file_type: FileTypePredicate,
+    file_content: Option<FileContent>,
+}
+
+impl Predicate<path::Path> for FileSystemEntryPredicate {
+    fn eval(&self, path: &path::Path) -> bool {
+        if self.file_type.is_file() {
+            if !self.file_type.eval(path) {
```
Comment (epage) [inline src/path/fs.rs]: I feel like it might be simpler if we just reused `FileType` and not `FileTypePredicate`

File: src/path/fs.rs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+                false
+            } else {
+                self.file_content.clone().unwrap() == FileContent::new(path).unwrap()
```
Comment (epage) [inline src/path/fs.rs]: `self.file_content.as_ref().unwrap()` addresses the `clone`

The first `unwrap` should probably be an `expect` so you can document your assumption of why it is always safe to `unwrap` in this context.

Should the second `unwrap`  instead have us return `false`?

File: src/path/fs.rs [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+            }
+        } else {
+            self.file_type.eval(path)
+        }
+    }
+}
+
+impl fmt::Display for FileSystemEntryPredicate {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "var is {}", self.file_type)
```
Comment (epage) [inline src/path/fs.rs]: Should it be `file_type` or the path?  The downside being we'd have to store off the path

File: src/path/fs.rs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+/// let predicate_file = predicate::path::eq_file_system_entry(Path::new("Cargo.toml"));
+/// assert_eq!(true, predicate_file.eval(Path::new("Cargo.toml")));
+/// assert_eq!(false, predicate_file.eval(Path::new("src")));
+/// assert_eq!(false, predicate_file.eval(Path::new("Cargo.lock")));
+///
+/// let predicate_dir = predicate::path::eq_file_system_entry(Path::new("src"));
+/// assert_eq!(false, predicate_dir.eval(Path::new("Cargo.toml")));
+/// assert_eq!(true, predicate_dir.eval(Path::new("src")));
+/// ```
+pub fn eq_file_system_entry(path: &path::Path) -> FileSystemEntryPredicate {
```
Comment (epage) [inline src/path/fs.rs]: Again, we can shorten this to `eq_entry` if it accepts multiple types f `eq_file` if it only works with files

**Comment by farodin91** [comment]
Fixed both request and the clippy failure
